src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -123,6 +131,25 @@ public boolean updateUserDescription(User user, String description) {
     return persistentStorageAgent.updateUserDescription(existingUser.getName(), description);
   }
 
+  /**
+   * Verify {@code user} exists in {@code users}.
+   * If so, set new description for {@code user} from persistent storage,
+   * and return true if the operation is successful.
+   * Otherwise, return false.
+   *
+   * @param user          user whose account to update
+   * @param isAdmin       new isAdmin
+   *
+   * @return true if {@code user} description gets updated. Otherwise, false.
+   */
+  public boolean updateUserIsAdmin(User user, boolean isAdmin) {
+    if (!isUserRegistered(user.getName())) {
+      return false;
+    }
```

Comment:
Is this check needed? Is there a way to create a user without going through registration?

Reply:
new User(...) ? We have non-registered users in the tests. Even if there is no where in our code that calls User constructor without registering it to userstore, it is safer to check isUserRegistered() first. Otherwise, there'll be NPE on line 150 (at existingUser.getName()).